server_test.go

```diff
@@ -4237,6 +4237,53 @@ func TestServerChunkedResponse(t *testing.T) {
 	}
 }
 
+func TestServerDisableBuffering(t *testing.T) {
+	t.Parallel()
+
+	expectedBody := bytes.Repeat([]byte("a"), 4096)
+
+	s := &Server{
+		Handler: func(ctx *RequestCtx) {
+			ctx.DisableBuffering()
+			ctx.SetStatusCode(StatusOK)
+			ctx.SetContentType("text/html; charset=utf-8")
+			reader := bytes.NewReader(expectedBody)
+			_, err := io.Copy(ctx, reader)
+			if err != nil {
+				t.Fatalf("Unexpected error when copying body: %v", err)
+			}
```

Comment:
This doesn't really test if it works. This test would still pass if you don't call `ctx.DisableBuffering()`. You might for example test the number of allocations, or you could pause while writing the body and check if the receiving end did receive it already.

Reply:
Just that check is not enough. I would like to see it block before the handler returns, then confirm the response body, and only when that has been received by the client return from the handler. That way you really check if it works the way it's supposed to work.